src/Login.js

```diff
@@ -88,12 +85,33 @@ class Login extends Component {
           {this.state.error && (
             <Alert bsStyle="danger">{this.state.error}</Alert>
           )}
-          <FieldGroup
-            inputType="button"
-            label=""
-            buttonText="Submit"
-            onClick={this.onSubmit}
-          />
+
+          {this.state.enter_edit && (
+            <div>
+              <FieldGroup
+                onClick={this.props.history.push('/', { isEditing: true })}
```

Comment:
This might be better to go over in person, but what you're doing here is setting `this.state.enter_edit` to `true` when `returnToMain()` is called, which then renders this `FieldGroup` component that automatically calls `this.props.history.push('/', { isEditing: true })`. And that's why it redirects.

But that's a really roundabout and hacky way of doing it. Instead of setting some state that renders a component that automatically calls the function to redirect, you can just call the function to redirect in `returnToMain`.